lib/src/widgets/fields/ds_input_phone.widget.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:blip_ds/blip_ds.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:get/get.dart';
+import 'package:mask_text_input_formatter/mask_text_input_formatter.dart';
+
+import '../../models/ds_country.model.dart';
+import '../ds_bottomsheet_countries.widget.dart';
+
+class DSInputPhone extends StatefulWidget {
+  final String? hintText;
+
+  const DSInputPhone({
+    super.key,
+    this.hintText,
+  });
+
+  @override
+  State<DSInputPhone> createState() => _DSInputPhoneState();
+}
+
+class _DSInputPhoneState extends State<DSInputPhone> {
+  final dropdownValue =
+      Rx<DSCountry>(DSBottomSheetCountries.listCountries.first);
+  final String mask1 = '(##) ####-#####';
+  final String mask2 = '#################';
+  final String mask3 = '(##) #####-####';
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: () {
+        FocusScope.of(context).requestFocus(FocusNode());
+      },
+      child: Container(
+        decoration: BoxDecoration(
+            border: Border.all(width: 1.0, color: DSColors.neutralMediumWave),
+            borderRadius: const BorderRadius.all(Radius.circular(8.0)),
+            color: DSColors.neutralLightSnow),
+        child: Row(
+          children: [
+            DSTertiaryButton(
+              leadingIcon: Obx(
+                () => SvgPicture.asset(
+                  'assets/svg/flags/${dropdownValue.value.flag}.svg',
+                  width: 22.0,
+                  height: 16.0,
+                  package: DSUtils.packageName,
+                ),
+              ),
+              trailingIcon: const Padding(
+                padding: EdgeInsets.only(left: 4.0),
+                child: Icon(
+                  DSIcons.arrow_down_outline,
+                  size: 16.0,
+                ),
+              ),
+              onPressed: () async {
+                final result = await DSBottomSheetCountries.show();
+                dropdownValue.value = result;
+                final mask = result.name == 'Brasil' ? mask1 : mask2;
```

Comment:
**Ignore this comment if you're going to use the [Phone Number](https://pub.dev/packages/phone_number) package.**

Currently, the input field is always getting masked by the Brazilian format (even though I had other country selected) and has the limit of 11 digits.

<img width="385" alt="image" src="https://github.com/takenet/blip-ds-flutter/assets/91232521/9f9df96f-dc82-4363-aad0-376da2b092b9">

Reply:
Done.